packages/jupyter-ai/src/components/chat.tsx

```diff
@@ -35,6 +35,26 @@ type ChatBodyProps = {
   focusInputSignal: ISignal<unknown, void>;
 };
 
+/**
+ * Determines the name of the current persona based on the message history.
+ * Defaults to `'Jupyternaut'` if history is insufficient.
+ */
+function getPersonaName(messages: AiService.ChatMessage[]): string {
+  let personaName = 'Jupyternaut';
+
+  for (let i = messages.length - 1; i >= 0; i--) {
```

Comment:
Comment for other reviewers and context: using a traditional for loop makes sense here. Unlike Python, there is no way in JS to get a reverse iterators for arrays so using a `for ... of` loop that could improve readability doesn't make sense here as it would require reversing the array first which is costly